---
title: Livestock Breeding and Genomics - Notebook 4
author: Peter von Rohr
date: '2023-10-05'
output: html_notebook
params:
  doctype:
    label: Document Type
    value: solution
    choices:
    - exercise
    - solution
    - notebook
  isonline:
    label: Online (y/n)
    value: true
    choices:
    - true
    - false
---



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Problem 1: Breeding Values For a Monogenic Trait
We assume that the absorption of cholesterol is determined by a certain enzyme. The level of enzyme production is determined by a single bi-allelic locus $E$. The genotype frequencies and the genotypic values for the two dairy cattle populations `Original Braunvieh` and `Brown Swiss` are given in the following table.

```{r table-geno-freq-value, echo=FALSE}
l_maf <- list(ob = 0.25, bs = 0.1)
l_a <- list(ob = 15, bs = 29)
l_d <- list(ob = 3, bs = 0)
tbl_geno_freq_value <- tibble::tibble(Variable = c(
  "$f(E_1E_1)$",
  "$f(E_1E_2)$",
  "$f(E_2E_2)$",
  "$a$",
  "$d$"),
  `Original Braunvieh` = c(
    l_maf$ob^2, 
    2*l_maf$ob*(1-l_maf$ob),
    (1-l_maf$ob)^2,
    l_a$ob,
    l_d$ob
  ),
  `Brown Swiss` = c(
    l_maf$bs^2,
    2*l_maf$bs * (1-l_maf$bs),
    (1-l_maf$bs),
    l_a$bs,
    l_d$bs
  ))
knitr::kable(tbl_geno_freq_value, booktabs = TRUE, longtable = TRUE, escape = FALSE)
```


### Hints
* Assume that allele $E_1$ is the allele with the positive effect on the enzyme level
* Assume that the Hardy-Weinberg Equilibrium holds in both populations


### Your Task
Compute the breeding values for all three genotypes in both populations. 


### Your Solution

* Use the formula shown in the class or in the course notes to compute the breeding values for the three genotypes
* Adapt the parameters that change for the two populations.





## Problem 2: Quantitative Genetics
In a population the following numbers of genotypes were counted for a given genetic locus called $A$. 

```{r GenotypeFreqs, echo=FALSE}
dfGenotypeFreq <- tibble::tibble(Genotypes = c("$A_1A_1$", "$A_1A_2$", "$A_2A_2$"),
                             Numbers   = c(24, 53, 23))
knitr::kable(dfGenotypeFreq, booktabs = TRUE, longtable = TRUE, escape = FALSE)
```

a) Compute the genotype frequencies
b) Compute the allele frequencies
c) Compute the population mean $\mu$ under the following assumptions

* the difference between the genotypic values of the homozygous genotypes is $20$ and
* the genotypic value of the heterozygous genotype is $2$.


### Your Solution
The requested quantities can be computed according to material presented during the lecture.



 


```{r, echo=FALSE, results='asis'}
cat('\n---\n\n _Latest Changes: ', format(Sys.time(), '%Y-%m-%d %H:%M:%S'), ' (', Sys.info()['user'], ')_\n', sep = '')
```
 
